Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -3327,13 +3404,14 @@ bool Parser::match_identifier() const
         if (m_state.current_token.value() == "yield"sv)
             return !m_state.strict_mode && !m_state.in_generator_function_context;
         if (m_state.current_token.value() == "await"sv)
-            return m_program_type != Program::Type::Module;
+            return m_program_type != Program::Type::Module && !m_state.in_async_function_context && !m_state.in_class_field_initializer;
         return true;
     }
 
     return m_state.current_token.type() == TokenType::Identifier
+        || m_state.current_token.type() == TokenType::Async
         || (m_state.current_token.type() == TokenType::Let && !m_state.strict_mode)
-        || (m_state.current_token.type() == TokenType::Await && m_program_type != Program::Type::Module)
+        || (m_state.current_token.type() == TokenType::Await && m_program_type != Program::Type::Module && !m_state.in_async_function_context && !m_state.in_class_field_initializer)
```

Comment:
Where does the condition `!m_state.in_class_field_initializer` come from? Looking at the spec I would say it is allowed?
If there is some reason can we maybe add a comment?
A case we now do wrong:
```js
await = 2;
class B {
    field = \u0061wait += 2;
    field2 = await;
}
console.log(new B().field2)
```
Which gives `4`.

Reply:
What I was trying to implement is:
https://tc39.es/ecma262/#sec-class-definitions-static-semantics-early-errors
`It is a Syntax Error if ClassStaticBlockStatementList Contains await is true.`
but I see that `in_class_field_initializer` is not only for static initializer blocks, is there a better field to go off?

Reply:
Hmm strange, we don't have any flag like that. You can just add one and it wouldn't surprise me if we have more of these problems.